mobile/lib/features/wallet/create_invoice_screen.dart

```diff
@@ -27,78 +27,140 @@ class _CreateInvoiceScreenState extends State<CreateInvoiceScreen> {
   Amount? amount;
   final TextEditingController _amountController = TextEditingController();
 
+  final _formKey = GlobalKey<FormState>();
+  bool showValidationHint = false;
+
+  final WalletService walletService = const WalletService();
+  final ChannelConstraintsService channelConstraintsService = const ChannelConstraintsService();
+
+  @override
+  void dispose() {
+    _amountController.dispose();
+    super.dispose();
+  }
+
   @override
   Widget build(BuildContext context) {
-    WalletInfo info = context.watch<WalletChangeNotifier>().walletInfo;
-    log("Refresh receive screen: ${info.balances.onChain}");
+    int channelCapacity = channelConstraintsService.getLightningChannelCapacity();
+    int totalMaxAmount = (channelCapacity / 2).floor();
+    int balance = context.watch<WalletChangeNotifier>().walletInfo.balances.lightning.sats;
+    int maxAmount = max(totalMaxAmount - balance, 0);
```

Comment:
can this actually be a negative number??? if this happened, then lower in the file we would display `0` in few places, where this would really confuse the user. e.g. "you can receive up to 0 sats".

(if this is a shortcut, then it's fine, but please add a comment about it ;D )

Reply:
You can receive up to `0` sats is correct in this context. I understand it can be confusing, but I would come back to it once we have more user feedback. 
This goes towards "explaining the channel boundaries to the user" - which is not easy.

Reply:
can we please add a comment in the code to reflect that? e.g.
`// it can go below 0 if the user has an unbalanced channel`

thanks 🙏